File: android/app/src/main/java/ru/maksimbulva/chess/mvp/BasePresenter.kt
package ru.maksimbulva.chess.mvp

import androidx.lifecycle.ViewModel

abstract class BasePresenter<V, VM> where VM : ViewModel {

    protected var attachedView: V? = null
    protected lateinit var viewModel: VM

    open fun onCreate(viewModel: VM) {
        this.viewModel = viewModel
    }

    open fun onAttachedView(view: V) {
        attachedView = view
    }

    open fun onStart() {
    }

    open fun onStop() {
    }

    open fun onDetachedView() {
        attachedView = null
    }

    open fun onDestroyed() {
    }
}

File: ChessApp/ChessApp/Mvp/BasePresenter.cs
using System;
using System.Reactive.Disposables;

namespace ChessApp.Mvp
{
    internal abstract class BasePresenter<V> where V : class, IView
    {
        protected V attachedView;

        protected readonly CompositeDisposable disposables = new CompositeDisposable();

        public virtual void OnCreate()
        {
        }

        public virtual void OnAttachedView(V view)
        {
            attachedView = view;
        }

        public virtual void OnStart() {
        }

        public virtual void OnStop() {
        }

        public virtual void OnDetachedView() {
            attachedView = null;
            disposables.Dispose();
            disposables.Clear();
        }

        public virtual void OnDestroyed() {
        }

        protected void AddSubscription(IDisposable disposable) {
            disposables.Add(disposable);
        }
    }
}
